packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -207,18 +196,22 @@ async function fetchLatestSnapshotCore(
             },
             async (event) => {
                 const startTime = performance.now();
-                const response = await snapshotDownloader(
-                    url,
-                    {
-                        body: postBody,
-                        headers,
-                        signal: controller?.signal,
-                        method: "POST",
-                    },
-                ) as IOdspResponse<IOdspSnapshot>;
+                const response = fetchBinarySnapshotFormat ? await fetchSnapshotContentsCoreV2(
```

Comment:
Can we push all this logic (including formatting of payload and parsing it) into snapshotDownloader callback, such that it is fully encapsulated there and no other layer (including this one) has to know anything about differences of formats or how many formats we support?
I.e. the code in fetchSnapshotCore() & prefetchLatestSnapshot() to be the only place where we have a fork and check what format we are using, all the rest of the code not having a clue about formats.
I'd go even further and refactor it into single function such that we do not need to duplicate code between fetchSnapshotCore() & prefetchLatestSnapshot() - they just get downloader from it and run with it

Reply:
I went further :)